src/player.cpp

```diff
@@ -2941,11 +2941,15 @@ void player::shout( std::string msg )
     }
 
     if( noise <= minimum_noise ) {
-        add_msg( m_warning, _( "The sound of your voice is almost completely muffled!" ) );
+        add_msg_player_or_npc( m_warning,
+                               _( "The sound of your voice is almost completely muffled!" ),
+                               _( "The sound of <npcname>'s voice is almost completely muffled!" ) );
         msg.clear();
     } else if( noise * 2 <= noise + penalty ) {
         // The shout's volume is 1/2 or lower of what it would be without the penalty
-        add_msg( m_warning, _( "The sound of your voice is significantly muffled!" ) );
+        add_msg_player_or_npc( m_warning,
+                               _( "The sound of your voice is significantly muffled!" ),
+                               _( "The sound of <npcname>'s voice is significantly muffled!" ) );
```

Comment:
I'm not sure if this is needed, and if it is it needs to be worked into the sound description somehow instead of printed here, because that handles whether the player can hear it at all or not.

Reply:
NPC says or shouts something, and the log provides us with info about it. I don't understand what's the problem.

Reply:
Because the player might not be aware of it.

For example the player is deaf and the npc is on 5he other side of a wall, we shouldn't print a message.